---
title: Is the protein folding problem still beyond us?
author: Arman Bilge
date: Thursday 4 June 2015
csl: apa.csl
references:
- id: Bia+14
  type: article-journal
  author:
  - given: Marco
    family: Biasini
  - given: Stefan
    family: Bienert
  - given: Andrew
    family: Waterhouse
  - given: Konstantin
    family: Arnold
  - given: Gabriel
    family: Studer
  - given: Tobias
    family: Schmidt
  - given: Florian
    family: Kiefer
  - given: Tiziano Gallo
    family: Cassarino
  - given: Martino
    family: Bertoni
  - given: Lorenza
    family: Bordoli
  - given: Torsten
    family: Schwede
  issued:
    year: 2014
  title: "SWISS-MODEL: modelling protein tertiary and quaternary structure using evolutionary information"
  container-title: Nucleic Acids Research
  volume: 42
  issue: Web Server
  page: W252–W258
  DOI: 10.1093/nar/gku340
- id: Dav+12
  type: article-journal
  author:
    - given: Aram
      family: Davtyan
    - given: Nicholas P.
      family: Schafer
    - given: Weihua
      family: Zheng
    - given: Cecilia
      family: Clementi
    - given: Peter G.
      family: Wolynes
    - given: Garegin A.
      family: Papoian
  issued:
    year: 2012
  title: "AWSEM-MD: Protein Structure Prediction Using Coarse-Grained Physical Potentials and Bioinformatically Based Local Structure Biasing"
  container-title: J. Phys. Chem. B
  volume: 116
  issue: 29
  page: 8494–8503
  DOI: 10.1021/jp212541y
- id: Fis09
  type: chapter
  author:
  - given: András
    family: Fiser
  issued:
    year: 2009
  title: Comparative Protein Structure Modelling
  container-title: From Protein Structure to Function with Bioinformatics
  editor:
  - given: Daniel John
    family: Rigden
  publisher: Springer
  publisher-place: Netherlands
  page: 57–90
  DOI: 10.1007/978-1-4020-9058-5_3
- id: Her+14
  type: article-journal
  author:
  - given: Joseph L.
    family: Herman
  - given: Christopher J.
    family: Challis
  - given: Ádám
    family: Novák
  - given: Jotun
    family: Hein
  - given: Scott C.
    family: Schmidler
  issued:
    year: 2014
  title: Simultaneous Bayesian Estimation of Alignment and Phylogeny under a Joint Model of Protein Sequence and Structure
  container-title: Mol Biol Evol
  volume: 31
  issue: 9
  page: 2251–2266
  DOI: 10.1093/molbev/msu184
- id: Hua+14
  type: article-journal
  author:
  - given: Yuanpeng J.
    family: Huang
  - given: Binchen
    family: Mao
  - given: James M.
    family: Aramani
  - given: Gaetano T.
    family: Montelione
  issued:
    year: 2014
  title: Assessment of template-based protein structure predictions in CASP10
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 46–56
  DOI: 10.1002/prot.24488
- id: Kal+12
  type: article-journal
  author:
  - given: Morten
    family: Källberg
  - given: Haipeng
    family: Wang
  - given: Sheng
    family: Wang
  - given: Jian
    family: Peng
  - given: Zhiyong
    family: Wang
  - given: Hui
    family: Lu
  - given: Jinbo
    family: Xu
  issued:
    year: 2012
  title: Template-based protein structure modeling using the RaptorX web server
  container-title: Nature Protocols
  volume: 7
  issue: 8
  page: 1511–1522
  DOI: 10.1038/nprot.2012.085
- id: Kav+98
  type: article-journal
  author:
  - given: Jeffrey S.
    family: Kavanaugh
  - given: Jamie A.
    family: Weydert
  - given: Paul H.
    family: Rogers
  - given: Arthur
    family: Arnone
  issued:
    year: 1998
  title: "High-Resolution Crystal Structures of Human Hemoglobin with Mutations at Tryptophan 37beta: Structural Basis for a High-Affinity T-State"
  container-title: Biochemistry
  volume: 37
  issue: 13
  page: 4358–4373
  DOI: 10.1021/bi9708702
- id: KFM14
  type: article-journal
  author:
  - given: Andriy
    family: Kryshtafovych
  - given: Krzysztof
    family: Fidelis
  - given: Andriy
    family: Kryshtafovych
  - given: John
    family: Moult
  issued:
    year: 2014
  title: CASP10 results compared to those of previous CASP experiments
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 164–174
  DOI: 10.1002/prot.24448
- id: Ko+12
  type: article-journal
  author:
  - given: Junsu
    family: Ko
  - given: Hahnbeom
    family: Park
  - given: Lim
    family: Heo
  - given: Chaok
    family: Seok
  title: GalaxyWEB server for protein structure prediction and refinement
  container-title: Nucleic Acids Research
  volume: 40
  issue: Web Server
  page: W294–W297
  DOI: 10.1093/nar/gks493
- id: Lup08
  type: article-journal
  author:
  - given: Andrei N.
    family: Lupas
  issued:
    year: 2008
  title: The long coming of computational structural biology
  container-title: Journal of Structural Biology
  volume: 163
  issue: 3
  page: 254–257
  DOI: 10.1016/j.jsb.2008.02.006
- id: LWZ09
  type: chapter
  author:
  - given: Jooyoung
    family: Lee
  - given: Sitao
    family: Wu
  - given: Yang
    family: Zhang
  issued:
    year: 2009
  title: "*Ab Initio* Protein Structure Prediction"
  container-title: From Protein Structure to Function with Bioinformatics
  editor:
  - given: Daniel John
    family: Rigden
  publisher: Springer
  publisher-place: Netherlands
  page: 3–25
  DOI: 10.1007/978-1-4020-9058-5_1
- id: Mar+11
  type: article-journal
  author:
  - given: Debora S.
    family: Marks
  - given: Lucy J.
    family: Colwell
  - given: Robert
    family: Sheridan
  - given: Thomas A.
    family: Hopf
  - given: Andrea
    family: Paganani
  - given: Riccardo
    family: Zecchina
  - given: Chris
    family: Sander
  issued:
    year: 2011
  title: Protein 3D Structure Computed from Evolutionary Sequence Variation
  container-title: PLoS ONE
  volume: 6
  issue: 12
  page: e28766
  DOI: 10.1371/journal.pone.0028766
- id: Mai+10
  type: article-journal
  author:
  - given: Gia G.
    family: Maisuradze
  - given: Patrick
    family: Senet
  - given: Cezary
    family: Czaplewski
  - given: Adam
    family: Liwo
  - given: Harold A.
    family: Scheraga
  issued:
    year: 2010
  title: Investigation of Protein Folding by Coarse-Grained Molecular Dynamics with the UNRES Force Field
  container-title: J. Phys. Chem.
  volume: 114
  issue: 13
  page: 4471–4485
  DOI: 10.1021/jp9117776
- id: MHS12
  type: article-journal
  author:
  - given: Debora S.
    family: Marks
  - given: Thomas A.
    family: Hopf
  - given: Chris
    family: Sander
  issued:
    year: 2012
  title: Protein structure prediction from sequence variation
  container-title: Nature Biotechnology
  volume: 30
  issue: 11
  page: 1072–1080
  DOI: 10.1038/nbt.2419
- id: MNF14
  type: article-journal
  author:
  - given: Vahid
    family: Mirjalili
  - given: Keenan
    family: Noyes
  - given: Michael
    family: Feig
  issued:
    year: 2014
  title: Physics-based protein structure refinement through multiple molecular dynamics trajectories and structure averaging
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 196–207
  DOI: 10.1002/prot.24336
- id: Moo65
  type: article-journal
  author:
  - given: Gordon E.
    family: Moore
  issued:
    year: 1965
  title: Cramming more components onto integrated circuits
  container-title: Electronics
  volume: 38
  issue: 8
  page: 114–117
- id: Mou+14
  type: article-journal
  author:
  - given: John
    family: Moult
  - given: Krzysztof
    family: Fidelis
  - given: Andriy
    family: Kryshtafovych
  - given: Torsten
    family: Schwede
  - given: Anna
    family: Tramontano
  issued:
    year: 2014
  title: Critical assessment of methods of protein structure prediction (CASP) — round x
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 1–6
  DOI: 10.1002/prot.24452
- id: NCJ14
  type: article-journal
  author:
  - given: Timothy
    family: Nugent
  - given: Domenico
    family: Cozzetto
  - given: David T.
    family: Jones
  issued:
    year: 2014
  title: Evaluation of predictions in the CASP10 model refinement category
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 98–111
  DOI: 10.1002/prot.24377
- id: NJ12
  type: article-journal
  author:
  - given: Timothy
    family: Nugent
  - given: David T.
    family: Jones
  issued:
    year: 2012
  title: Accurate de novo structure prediction of large transmembrane protein domains using fragment-assembly and correlated mutation analysis
  container-title: PNAS
  volume: 109
  issue: 24
  page: E1540–E1547
  DOI: 10.1073/pnas.1120036109
- id: Rav+12
  type: article-journal
  author:
  - given: Alpan
    family: Raval
  - given: Stefano
    family: Piana
  - given: Michael P.
    family: Eastwood
  - given: Ron O.
    family: Dror
  - given: David E.
    family: Shaw
  issued:
    year: 2012
  title: Refinement of protein structure homology models via long, all-atom molecular dynamics simulations
  container-title: Proteins
  volume: 80
  issue: 8
  page: 2071–2079
  DOI: 10.1002/prot.24098
- id: SW86
  type: article-journal
  author:
  - given: Robert H.
    family: Swendsen
  - given: Jian-Sheng
    family: Wang
  issued:
    year: 1986
  title: Replica Monte Carlo Simulation of Spin-Glasses
  container-title: Phys. Rev. Lett.
  volume: 57
  issue: 21
  page: 2607–2609
  DOI: 10.1103/PhysRevLett.57.2607
- id: Tai+14
  type: article-journal
  author:
  - given: Chin-Hsien
    family: Tai
  - given: Hongjun
    family: Bai
  - given: Todd J.
    family: Taylor
  - given: Byungkook
    family: Lee
  issued:
    year: 2014
  title: Assessment of template-free modeling in CASP10 and ROLL
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 57–83
  DOI: 10.1002/prot.24470
- id: XZ12
  type: article-journal
  author:
  - given: Dong
    family: Xu
  - given: Yang
    family: Zhang
  issued:
    year: 2012
  title: "*Ab initio* protein structure assembly using continuous structure fragments and optimized knowledge-based force field"
  container-title: Proteins
  volume: 80
  issue: 7
  page: 1715–1735
  DOI: 10.1002/prot.24065
- id: Yan+15
  type: article-journal
  author:
  - given: Jianyi
    family: Yang
  - given: Renxiang
    family: Yan
  - given: Ambrish
    family: Roy
  - given: Dong
    family: Xu
  - given: Jonathan
    family: Poisson
  - given: Yang
    family: Zhang
  issued:
    year: 2015
  title: "The I-TASSER Suite: protein structure and function prediction"
  container-title: Nature Methods
  volume: 12
  issue: 1
  page: 7–8
  DOI: 10.1038/nmeth.3213
- id: ZS11
  type: article-journal
  author:
  - given: Hongyi
    family: Zhou
  - given: Jeffrey
    family: Skolnick
  issued:
    year: 2011
  title: "GOAP: A Generalized Orientation-Dependent, All-Atom Statistical Potential for Protein Structure Prediction"
  container-title: Biophysical Journal
  volume: 101
  issue: 8
  pages: 2043–2052
  DOI: 10.1016/j.bpj.2011.09.012
...

\frenchspacing

Proteins play a fundamental role in effectively all metabolic pathways and thus
are at the heart of molecular biology and biochemistry.
The function of a protein is directly related to its structure, determined by
the series of folds that occur along the sequence of amino acids comprising the
protein [@Fis09].
Knowledge of protein structure also has important implications for evolutionary
biology.
Specifically, because structure is conserved more strongly than molecular
sequence [@Fis09], structural information makes it possible to accurately
determine the evolutionary relationships between distantly related species
[@Her+14].
However, the technical difficulties in experimentally determining the structure
of a protein have created a rapidly-growing gap between the total number of
sequenced proteins and those with a known structure [@LWZ09; @MHS12].
@LWZ09 suggested that structural prediction *in silico* is the only way to
close this gap.

@Fis09 summarised the computational approaches to protein structure prediction
as using either the laws of physics (*ab initio* methods) or the theory of
evolution (comparative, template-based methods).
However, @Lup08 emphasised that the true solution to this problem is *ab
initio* given the limited usefulness of comparative methods to make accurate
predictions, especially in the absence of substantial prior information.
It is worth noting that this dichotomy is less clear-cut in practice: many
otherwise *ab inito* methods do take advantage of the knowledge in databases
[@Fis09; @LWZ09; @ZS11; @XZ12] and template-based methods still rely on physics
for refinement as it is impossible to make accurate predictions solely based on
homology [@Fis09; @Rav+12].

The ideal approach to structure prediction would be full simulation of protein
synthesis and folding at the atomic level relying only on quantum mechanics and
the fundamental laws of physics [@LWZ09].
Furthermore, such simulations could yield important insights into the folding
process itself, which is equally complex and varies even between proteins with
homologous structure [@Lup08].
However, due to the immense computational resources necessary for its
implementation, little serious effort has been made in this area [@LWZ09].
Proteins are generally modelled on the atomic level, where each atom is
represented by three coordinates (either using the Cartesian system or the
torsion-angle system), such that a protein model consisting of $n$ atoms has
$3\left(n-1\right)$ degrees of freedom [@XZ12].
For example, the beta subunit of haemoglobin consists of nearly 5000 atoms for
some 15000 degrees of freedom [@Kav+98].
The number of atoms also provides a lower bound on the number of particles
considered in an accurate physical model, which must also model the solvent.
Unfortunately it is infeasible to complete the calculations necessary for a
complete simulation on a realistic time scale.

This limitation forces the use of various approximations in the form of
mathematical, statistical, and computational techniques.
From this perspective, the protein structure prediction problem is framed
as an optimisation problem where the key challenges are deriving a scoring
function that gives the best score to the true structure and developing a
search algorithm that can find this structure with the optimal score [@LWZ09].
The correct structure is assumed to be the most thermodynamically stable one
and thus structures are scored according to an energy function [@Fis09;
@LWZ09].
Both physics- and knowledge-based energy functions are used for this purpose
[@LWZ09; @ZS11].

To make computations feasible, an energy function should ideally reduce the
number of independent particles being considered.
Modelling the protein *in vacuo* is an easy but poor-performing approximation
[@LWZ09] due to the significance of water-mediated interactions in protein
folding [@Dav+12].
Instead, a coarse-grained force field can systematically reduce the number of
particles in the model without ignoring important interactions by using the
average energy over the neglected degrees of freedom
[@Mai+10].
Each residue is generally represented by the atoms in its backchain (or some
approximation, as in the UNRES model) and the centre of mass of its side chain
[@LWZ09; @Dav+12].
Both the UNRES [@Mai+10] and AWSEM [@Dav+12] models are examples of
coarse-grained energy functions that are primarily physics-based but are
augmented with knowledge from the PDB (although a later development of UNRES
replaced these knowledge-based terms with additional physics-based terms).
The AWSEM energy function includes terms for the bond lengths and angles of the
backbone, the placement of the side chain, the interactions between neighbouring
amino acid residues and water, and the propensity for a particular residue to
be within or on the outside of the protein [@Dav+12].

Alternatively, the GOAP, QUARK, and I-TASSER energy functions are primarily
knowledge-based [@ZS11; @XZ12; @Yan+15].
The information for these methods comes from structural regularities as
determined empirically from known experimental structures in the PDB
[@XZ12; @Yan+15].
Notably, all of the terms used in QUARK are in fact physically-motivated
despite being computed statistically [@XZ12].
The EVfold method, which also uses a statistical energy function, takes a
radically different approach [@Mar+11, @NJ12].
@Mar+11 attempted to infer the proximity of residues in the structure by using
information from the mutations across a protein family.
Specifically, they assumed that residues with physical proximity show higher
degrees of coevolution, observed as correlated mutations in the sequence
alignment [@Mar+11].
Using this data, predicting the protein structure simplified to identifying the
signal from the noise and deriving an energy function that reflected how well
a given structure supported the observed residue correlations.

With an energy function in hand, it remains only to find the best model.
Parameter optimisation is a well-studied problem with several existing
methodologies.
The main techniques used for protein structure prediction include Monte Carlo
(with simulated annealing) [@Fis09; @LWZ09; @NJ12; @XZ12; @Yan+15] and
molecular dynamics simulations
[@Lup08; @Fis09; @LWZ09; @Mar+11; @Dav+12; @Rav+12; @MNF14; @Mai+10].
Molecular dynamics strategies are especially suited for structure refinement
[@Rav+12].
Other optimisation strategies, such as genetic algorithms, have been applied
to structure prediction [@LWZ09; @Fis09] but appear to be less popular or
successful [@Mou+14].
Searching for the optimal structure is difficult because the "landscape"
defined by the energy function is rugged, causing algorithms to become stuck
in local minima [@LWZ09].
Search strategies benefit from an approximate model of the protein stucture
that reduce the degrees of freedom and, therefore, the total space of
structures to be searched [@XZ12].

Monte Carlo and molecular dynamics algorithms differ in the amount of change
that occurs in a single iteration and are thus best applied in different
situations.
In a molecular dynamics simulation, the protein structure at any given time
can be found by solving Newton's equations of motions, with the idea that
after enough time the protein will be in its most thermodynamically stable
state [@LWZ09].
However, because there is no explicit solution, numerical integration must be
used instead.
Thus, each iteration of the simulation is generally on the order of
femtoseconds (one quadrillionth of a second) [@LWZ09].
Given that a complete simulation takes milliseconds, trillions of iterations
of the molecular dynamics algorithm become necessary [@LWZ09].
Each iteration involves computing the derivative, or slope, of the energy
function at that point with respect to every degree of freedom.
This requirement restricts either the complexity of the energy function,
as it must be analytically differentiable, or increases the computational
effort if numerical differentiation is used instead.
For the above reasons, molecular dynamics simulations are best-suited for
refinement of a fairly-accurate structure using a physics-based energy
function [@LWZ09].

On the other hand, Monte Carlo algorithms use large proposals to rapidly
move through the model space.
The design of these moves is critical to the algorithm's efficiency.
QUARK, for example, implements 11 moves that operate at the residue-, segment-,
and topology-level [@XZ12].
Using these moves, the Monte Carlo algorithm samples structures according to
the energy function at a given temperature.
The temperature is gradually lowered so that the algorithm eventually settles
at a single, optimal structure [@LWZ09].
This procedure is known as simulated annealing [@LWZ09].
Although Monte Carlo with simulated annealing is not a greedy algorithm due
to its use of the Metropolis criterion, if the algorithm is exploring a local
optimum when the temperature is decreased then it will very likely get stuck at
that local optimum.
Replica-exchange is a flavour of Monte Carlo that builds on simulated annealing
and is used in QUARK, I-TASSER, and Rosetta [@SW86; @XZ12; @Yan+15].
Unlike the standard Monte Carlo algorithm, the replica-exchange algorithm
involves simultaneous simulation of several replicas of the model at different
temperatures [@SW86], with more replicas used for larger proteins [@Yan+15].
Periodically, a global swap of model features between the replicas is
attempted [@SW86].
The use of multiple simulations and exchanges between these simulations helps
to prevent the algorithm from getting stuck in a local optimum.
Furthermore, Monte Carlo algorithms are indifferent to the type of energy
function used and, if implemented well, can converge to an optimal structure
fairly quickly.

But even with sophisticated search algorithms, finding the optimal structure
remains very difficult.
Template-based prediction aggressively reduces the search space by applying
the assumption that a small change in sequence should only cause a small change
in structure [@Fis09].
The first steps for these methods involve searching through the PDB for
homologous structures and deciding which parts of these to use, if any
[@Fis09].
This requires the creation of an accurate sequence alignment of the target to
the templates [@Fis09].
There are many existing, well-developed search tools, such as PSI-BLAST, that
facilitate this process [@Fis09].
Specialised tools can take advantage of additional structural information when
creating the alignment: for example, by avoiding placing gaps in important
structural elements and checking for concurrence between the known and
predicted secondary structures [@Fis09; @Bia+14].
The RaptorX server attempts to maximise the consistency between all
target-template pairs using multiple-template threading and also models
uncertainties in the final alignment using probability [@Kal+12].
This step is particularly important because the accuracy of the final,
predicted structure is sensitive to the accuracy of the sequence alignment
[@Fis09].
The choice of a particular template over other options also depends on its
alignment.
Choosing which templates to use for which sections of the target protein is
challenging to do automatically and is ideally overseen by experts [@Fis09].
Furthermore, @KFM14 indicated that selecting the best available template (i.e.,
those with closest resemblance to the target structure) becomes more difficult
when there are a large number of options.
Again, the use of additional structural information (as discussed above) can
help choose between templates [@Bia+14].
Once the templates are selected, the model is assembled using some combination
of the known structures [@Fis09].
Any improvements on this model are done via the *ab inito* techniques,
hence my classification of template-based modelling as primarily a search
strategy.

A shortcoming of template-based prediction is that it neglects non-homologous
loops and insertions in the protein, which are often important to its function
[@Fis09].
These can be predicted *ab initio* using the methods described previously.
However, it turns out that the structure of loops can be also be predicted by
using similar fragments appearing in other proteins on the PDB [@Fis09; @XZ12].
The small number of unique loops occurring in the PDB indicates that different
proteins reuse the same fragments and means that they can be used to predict
the structure of loops in the target protein [@Fis09].
This observation has led to a greater effort to classify and improve search
algorithms for loop fragments, as this is the primary limiting factor [@Fis09].
However, *ab initio* methods are still necessary for longer, unrepresented
fragments.
The use of fragment searches can also be extended from loops to modelling any
part of the protein structure.
@XZ12 successfully combined the use of fragments with the Monte Carlo algorithm
in QUARK.

It is important to note that no search algorithm can compensate for a poor
scoring function, as its responsibility is only to find the best-scoring
structure by that function.
Improvements to the search algorithm only affect the speed of prediction and
the probability that the best-scoring structure is found and not necessarily
that the accuracy of the prediction improves [@MNF14].
Furthermore, @Rav+12 suggested that inaccurate force fields are the limiting
factor in protein structure prediction because they found that refinement using
molecular dynamics simulations caused the model to move away from the true
structure.
Accordingly, most of the recent advances in protein structure prediction are
primarily focused on improving the energy functions used to score and rank
structures.

There is little doubt that significant progress has been made developing new
and creative methods for protein structure prediction, but has there been
equivalent progress in the accuracy of these predictions?
The CASP experiments are a biennial measure of advances in protein structure
prediction; the latest published results were for the tenth round [@Mou+14].
In the last decade of CASP, the best prediction methods improved by 10%,
although the bulk of this improvement was in the first five years
[@KFM14; @Mou+14].
These improvements are best seen in the modelling of small and simple
structures [@KFM14].
QUARK was identified as the best-performing method, folding proteins with up to
1.30 Å root mean squared deviation [@Hua+14; @XZ12], but overall no group or
method dominated in prediction accuracy [@Tai+14].
The refinement of predictions is a recent area of notable advances thanks to a
method that consistently improved the accuracy of structural predictions
[@Mou+14; @KFM14].
Interestingly, only in this latest round did methods using physics-based
energy functions have comparable performance to those using statistical energy
functions [@NCJ14].

It is clear that no matter what approach is taken to the protein structure
prediction problem there is still substantial progress to be made.
Although structure prediction methods are performing better than ever, they
still struggle to scale for larger proteins [@Mou+14; @Tai+14].
Additionally, accurately evaluating advances in protein structure is
challenging because it is difficult to establish how much of the recent
progress in is truly due to intellectual achievements in method design
[@KFM14].
Both the amount of computational power available and the number of
experimentally determined structures have increased substantially in the last
decade such that, even after controlling for these advances, it is still
difficult to perform a fair comparison [@KFM14].
Although this is promising in the sense that we will inevitably solve the
protein structure problem thanks to Moore's law [@Moo65], thus facilitating
research across the biological sciences, many scientists will remain
unsatisfied until we have true, intuitive understanding of the protein folding
process.

# References
