---
title: '[Is the protein folding problem still beyond us?]'
author: Arman Bilge
date: Friday 8 May 2015
csl: apa.csl
references:
- id: Dav+12
  type: article-journal
  author:
    - given: Aram
      family: Davtyan
    - given: Nicholas P.
      family: Schafer
    - given: Weihua
      family: Zheng
    - given: Cecilia
      family: Clementi
    - given: Peter G.
      family: Wolynes
    - given: Garegin A.
      family: Papoian
  issued:
    year: 2012
  title: AWSEM-MD: Protein Structure Prediction Using Coarse-Grained Physical Potentials and Bioinformatically Based Local Structure Biasing
  container-title: J. Phys. Chem. B
  volume: 116
  issue: 29
  page: 8494–8503
  DOI: 10.1021/jp212541y
- id: Fis09
  type: chapter
  author:
  - given: András
    family: Fiser
  issued:
    year: 2009
  title: Comparative Protein Structure Modelling
  container-title: From Protein Structure to Function with Bioinformatics
  editor:
  - given: Daniel John
    family: Rigden
  publisher: Springer
  publisher-place: Netherlands
  page: 57–90
  DOI: 10.1007/978-1-4020-9058-5_3
- id: Her+14
  type: article-journal
  author:
  - given: Joseph L.
    family: Herman
  - given: Christopher J.
    family: Challis
  - given: Ádám
    family: Novák
  - given: Jotun
    family: Hein
  - given: Scott C.
    family: Schmidler
  issued:
    year: 2014
  title: Simultaneous Bayesian Estimation of Alignment and Phylogeny under a Joint Model of Protein Sequence and Structure
  container-title: Mol Biol Evol
  volume: 31
  issue: 9
  page: 2251–2266
  DOI: 10.1093/molbev/msu184
- id: Kav+98
  author:
  - given: Jeffrey S.
    family: Kavanaugh
  - given: Jamie A.
    family: Weydert
  - given: Paul H.
    family: Rogers
  - given: Arthur
    family: Arnone
  type: article-journal
  issued:
    year: 1998
  title: "High-Resolution Crystal Structures of Human Hemoglobin with Mutations at Tryptophan 37beta: Structural Basis for a High-Affinity T-State"
  container-title: Biochemistry
  volume: 37
  issue: 13
  page: 4358–4373
  DOI: 10.1021/bi9708702
- id: KFM14
  type: article-journal
  author:
  - given: Andriy
    family: Kryshtafovych
  - given: Krzysztof
    family: Fidelis
  - given: Andriy
    family: Kryshtafovych
  - given: John
    family: Moult
  issued:
    year: 2014
  title: CASP10 results compared to those of previous CASP experiments
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 164–174
  DOI: 10.1002/prot.24448
- id: Lup08
  type: article-journal
  author:
  - given: Andrei N.
    family: Lupas
  issued:
    year: 2008
  title: The long coming of computational structural biology
  container-title: Journal of Structural Biology
  volume: 163
  issue: 3
  page: 254–257
  DOI: 10.1016/j.jsb.2008.02.006
- id: LWZ09
  type: chapter
  author:
  - given: Jooyoung
    family: Lee
  - given: Sitao
    family: Wu
  - given: Yang
    family: Zhang
  issued:
    year: 2009
  title: "*Ab Initio* Protein Structure Prediction"
  container-title: From Protein Structure to Function with Bioinformatics
  editor:
  - given: Daniel John
    family: Rigden
  publisher: Springer
  publisher-place: Netherlands
  page: 3-25
  DOI: 10.1007/978-1-4020-9058-5_1
- id: Mar+11
  type: article-journal
  author:
  - given: Debora S.
    family: Marks
  - given: Lucy J.
    family: Colwell
  - given: Robert
    family: Sheridan
  - given: Thomas A.
    family: Hopf
  - given: Andrea
    family: Paganani
  - given: Riccardo
    family: Zecchina
  - given: Chris
    family: Sander
  issued:
    year: 2011
  title: Protein 3D Structure Computed from Evolutionary Sequence Variation
  container-title: PLoS ONE
  volume: 6
  issue: 12
  page: e28766
  DOI: 10.1371/journal.pone.0028766
- id: MHS12
  type: article-journal
  author:
  - given: Debora S
    family: Marks
  - given: Thomas A
    family: Hopf
  - given: Chris
    family: Sander
  issued:
    year: 2012
  title: Protein structure prediction from sequence variation
  container-title: Nature Biotechnology
  volume: 30
  issue: 11
  page: 1072–1080
  DOI: 10.1038/nbt.2419
- id: MNF13
  type: article-journal
  author:
  - given: Vahid
    family: Mirjalili
  - given: Keenan
    family: Noyes
  - given: Michael
    family: Feig
  issued:
    year: 2014
  title: Physics-based protein structure refinement through multiple molecular dynamics trajectories and structure averaging
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 196–207
  DOI: 10.1002/prot.24336
- id: Mou+14
  type: article-journal
  author:
  - given: John
    family: Moult
  - given: Krzysztof
    family: Fidelis
  - given: Andriy
    family: Kryshtafovych
  - given: Torsten
    family: Schwede
  - given: Anna
    family: Tramontano
  issued:
    year: 2014
  title: Critical assessment of methods of protein structure prediction (CASP) — round x
  container-title: Proteins
  volume: 82
  issue: Suppl 2
  page: 1–6
  DOI: 10.1002/prot.24452
- id: NJ12
  type: article-journal
  author:
  - given: Timothy
    family: Nugent
  - given: David T.
    family: Jones
  issued:
    year: 2012
  title: Accurate de novo structure prediction of large transmembrane protein domains using fragment-assembly and correlated mutation analysis
  container-title: PNAS
  volume: 109
  issue: 24
  page: E1540–E1547
  DOI: 10.1073/pnas.1120036109
- id: Rav+12
  type: article-journal
  author:
  - given: Alpan
    family: Raval
  - given: Stefano
    family: Piana
  - given: Michael P.
    family: Eastwood
  - given: Ron O.
    family: Dror
  - given: David E.
    family: Shaw
  issued:
    year: 2012
  title: Refinement of protein structure homology models via long, all-atom molecular dynamics simulations
  container-title: Proteins
  volume: 80
  issue: 8
  page: 2071–2079
  DOI: 10.1002/prot.24098
- id: XZ12
  type: article-journal
  author:
  - given: Dong
    family: Xu
  - given: Yang
    family: Zhang
  issued:
    year: 2012
  title: "*Ab initio* protein structure assembly using continuous structure fragments and optimized knowledge-based force field"
  container-title: Proteins
  volume: 80
  issue: 7
  page: 1715–1735
  DOI: 10.1002/prot.24065
...

\frenchspacing

Proteins play a fundamental role in effectively all metabolic pathways and thus
are at the heart of molecular biology and biochemistry.
The function of a protein is directly related to its structure, the series of
folds that occur along the sequence of amino acids comprising the protein
[@Fis09].
Knowledge of protein structure also has important implications for evolutionary
biology.
Specifically, because structure is conserved more strongly than molecular
sequence [@Fis09], structural information makes it possible to accurately
determine the evolutionary relationships between distantly related species
[@Her+14].
However, the technical difficulties in experimentally determining the structure
of a protein have created a rapidly growing gap between the total number of
sequenced proteins and those with a known structure [@LWZ09; @MHS12].
@LWZ09 suggested that structural prediction *in silico* is the only way to
close this gap.

@Fis09 summarised the computational approaches to protein structure prediction
as using either the laws of physics (*ab initio* methods) or the theory of
evolution (comparative, template-based methods).
However, @Lup08 emphasised that the true solution to this problem is *ab
initio* given the limited usefulness of comparative methods to make accurate
predictions, especially in the absence of substantial prior information.
It is worth noting that the dichotomy described by @Fis09 is less clear-cut in
practice: many otherwise *ab inito* methods do take advantage of the knowledge
in databases [@Fis09; @LWZ09] and template-based methods still rely on physics
for refinement as it is impossible to make accurate predictions solely based on
homology [@Fis09].

The ideal approach to structure prediction would be naïve complete
simulation of protein synthesis and folding at the atomic level relying only on
quantum mechanics and the fundamental laws of physics [@LWZ09].
Furthermore, such simulations could yield important insights into the folding
process itself, which is equally complex and varies even between proteins with
homologous structure [@Lup08].
However, due to the immense computational resources necessary for its
implementation, little serious effort has been made in this area [@LWZ09].
These limitations force the use of various approximations in the form of
mathematical, statistical, and computational techniques.

From this perspective, the protein structure prediction problem is framed
as an optimisation problem, where they key challenges are deriving a scoring
function that gives the best score to the true structure and developing a
search algorithm that can find this structure with the optimal score [@LWZ09].
The correct structure is assumed to be the most thermodynamically stable one
and thus structures are scored according to an energy function [@Fis09;
@LWZ09].
Both physics- and knowledge-based energy functions can used for this purpose
[@LWZ09].

Given a scoring function, parameter optimisation is a well-studied problem with
several existing methodologies.
The main techniques applied to protein structure prediction include Monte
Carlo (with simulated annealing), molecular dynamics simulations, genetic
algorithms, and database searches [@Fis09; @LWZ09].



It is important to note that no search algorithm can compensate for a poor
scoring function, as its responsibility is to find the best-scoring structure
by that function.
Improvements to the search algorithm only affect the speed of prediction and
the probability that the best-scoring structure is found and not necessarily
that the accuracy of the prediction improves.
Given the exten
Most of the recent advances in protein structure prediction are primarily
focused on

The total search space for any reasonably-sized protein is immense.
Proteins are generally modelled on the atomic level, where each atom is
represented by three coordinates (either using the Cartesian system or the
torsion-angle system), such that a protein consisting of $n$ atoms has
$3\left(n-1\right)$ degrees of freedom [@XZ12].
For example, the beta subunit of haemoglobin consists of nearly 5000 atoms for
some 15000 degrees of freedom [@Kav+98].
The number of atoms also provides a lower bound on the number of particles
considered in an accurate physical model, which must also consider the solvent.

It comes as no surprise that template-based methods have been successful given
their ability to aggressively reduce the search space by applying the concept
that a small change in sequence should cause only a small change in structure
[@Fis09].
The first steps for these methods involve searching through the PDB for
homologous structures and deciding which parts of these to use, if any
[@Fis09].

It is clear that no matter what approach is taken to the protein structure
prediction problem,

Furthermore, it is difficult to establish how much of the recent advances in
template-based approaches are due to actual improvements in methodology versus


# References
